scxml_core/src/state_machine.cpp

```diff
@@ -475,11 +475,17 @@ TransitionResult StateMachine::executeAction(const Action& action)
         {
           return true;  // no precondition
         }
-        precond_res = precond_callbacks_[st](action);
-        if (!precond_res)
+        for (auto cb : precond_callbacks_[st])
         {
-          precond_res.msg = boost::str(boost::format("Precondition for state %s failed: %s") % st % precond_res.msg);
+          precond_res = cb(action);
+          if (!precond_res)
+          {
+            // break on the first failed precondition
+            precond_res.msg = boost::str(boost::format("Precondition for state %s failed: %s") % st % precond_res.msg);
```

Comment:
This would be more informative if it printed the index of the precondition that failed.

Reply:
Added!